errors.go

```diff
@@ -0,0 +1,8 @@
+package s3
+
+import "errors"
+
+var (
+	// ErrInvalidEncryptionCustomerKey will be returned while encryption customer key is invalid.
+	ErrInvalidEncryptionCustomerKey = errors.New("invalid encryption customer key")
```

Comment:
Maybe we can use `ErrServerSideEncryptionCustomerKey` here? As the input pair is called `ServerSideEncryptionCustomerKey`.